app/screens/channel/channel_title/index.tsx

```diff
@@ -0,0 +1,279 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {withDatabase} from '@nozbe/watermelondb/DatabaseProvider';
+import withObservables from '@nozbe/with-observables';
+import React from 'react';
+import {Text, TouchableOpacity, View} from 'react-native';
+
+import ChannelIcon from '@components/channel_icon';
+import CompassIcon from '@components/compass_icon';
+import FormattedText from '@components/formatted_text';
+import {General} from '@constants';
+import {MM_TABLES} from '@constants/database';
+import {useTheme} from '@context/theme';
+import {makeStyleSheetFromTheme} from '@utils/theme';
+import {t} from '@utils/i18n';
+
+import type {Database} from '@nozbe/watermelondb';
+
+import type ChannelInfoModel from '@typings/database/models/servers/channel_info';
+import type ChannelModel from '@typings/database/models/servers/channel';
+import type MyChannelSettingsModel from '@typings/database/models/servers/my_channel_settings';
+import type UserModel from '@typings/database/models/servers/user';
+
+const ConnectedChannelTitle = ({channel, onPress, canHaveSubtitle, currentUserId, channelInfo, channelSettings, teammateId, teammate}: ChannelTitleProps) => {
+    const theme = useTheme();
+
+    const style = getStyle(theme);
+    const channelType = channel.type;
+    const currentChannelName = channel ? channel.displayName : '';
```

Comment:
it looks like channel is mandatory and this is a dupe of line 31

Reply:
Corrected